### PR TITLE
chore: handle currupted message in rendezvous

### PR DESCRIPTION
**Pull request by vpavlin**
We have noticed a crash in nwaku using nim-libp2p rendezvous when an incompatible message was sent to the rendezvous protocol (see go-waku using older version of the spec https://github.com/berty/go-libp2p-rendezvous/blob/10a9ad4472b03029ee41f0b9b3096178f2ed0041/pb/rendezvous.proto#L68)

This PR adds checks to prevent the crash and handle the corrupted message gracefully

### CODE REVIEW

File: libp2p/protocols/rendezvous.nim
Comment (Menduist) [reply libp2p/protocols/rendezvous.nim:466]:
```suggestion
        decodedMsg = Message.decode(buf)
        msgRecv = decodedMsg.valueOr(raise RendezVousError.newException("Invalid message"))
```
For instance (didn't test)

Comment (vpavlin) [reply libp2p/protocols/rendezvous.nim:466]: Hmm, I don't think this would work since `decode` returns `Option`, not `Result` and I don't see `valueOr` on Option type, only `isNone`/`isSome`

Comment (Menduist) [reply libp2p/protocols/rendezvous.nim:466]: Nice, adding to the confusion.. We can merge this PR if you want, but this file also requires a overhaul to switch to Opt & fix all of theses issues (a lot of `get()` snuck in there)

I've started the bigger cleanup here: https://github.com/status-im/nim-libp2p/pull/902
I'll let you merge this PR to unblock you, or close it and wait for the bigger refacto :)

**Review by Menduist — APPROVED** [review]

File: libp2p/protocols/rendezvous.nim
Comment (vpavlin) [reply libp2p/protocols/rendezvous.nim:466]: As much as I'd love to get a commit in nim-libp2p it probably does not make sense with #902 on the way:)

**Comment by Menduist** [comment]
#902 got merged

**Pull request closed without merging**
